### PR TITLE
[RISCV][GISel] Remove s32 support on RV64 for DIV, and REM.

### PR DESCRIPTION
**Pull request by topperc**
Based on experience with SelectionDAG and experimental-rv64-legal-i32, I don't believe making s32 a legal type is viable without introducing an invariant that s32 values are always sign extended like Mips64 does. Mips64 does this with a separate 32-bit register class.

`experimental-rv64-legal-i32` is proposed for removal in ##102509.

This patch is part of a series to remove s32 support so we can remove the isel patterns that SelectionDAG is no longer using. To restore code quality, we will need to add custom W nodes like SelectionDAG.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/GlobalISel/alu-roundtrip.ll [hunk 1]
```diff
@@ -348,7 +348,9 @@ define i32 @sdiv_i32(i32 %a, i32 %b) {
 ;
 ; RV64IM-LABEL: sdiv_i32:
 ; RV64IM:       # %bb.0: # %entry
-; RV64IM-NEXT:    divw a0, a0, a1
+; RV64IM-NEXT:    sext.w a0, a0
+; RV64IM-NEXT:    sext.w a1, a1
+; RV64IM-NEXT:    div a0, a0, a1
```
Comment (michaelmaitland) [inline llvm/test/CodeGen/RISCV/GlobalISel/alu-roundtrip.ll:353]: For now, should we look one level down to see if we know operands are sign extended?

Comment (michaelmaitland) [reply llvm/test/CodeGen/RISCV/GlobalISel/alu-roundtrip.ll:353]: What if we look at it during custom legalization?

Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/alu-roundtrip.ll:353]: > Edit: I think thats what you mean by custom W nodes?

Yes. In order to do something during custom legalization we would need to convert it to a custom opcode.

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Pull request merged**